images/router/haproxy/conf/haproxy-config.template

```diff
@@ -318,7 +318,7 @@ backend be_edge_http:{{$cfgIdx}}
     {{- range $serviceUnitName, $weight := $cfg.ServiceUnitNames }}
       {{- if ne $weight 0 }}
         {{- with $serviceUnit := index $.ServiceUnits $serviceUnitName }}
-          {{- range $idx, $endpoint := endpointsForAlias $cfg $serviceUnit }}
+          {{- range $idx, $endpoint := randomEndpointsForAlias $cfg $serviceUnit true }}
```

Comment:
I think you need to do this conditionally based on a variable set above when haproxy.router.openshift.io/disable_cookies is considered.